### PR TITLE
[ticket/13306] add error level to collector

### PR DESCRIPTION
**Pull request by omniError**
https://tracker.phpbb.com/browse/PHPBB3-13306

PHPBB3-13306

### CODE REVIEW

**Comment by omniError** [comment]
@bantu is anything else keeping this from being merged? thanks

File: phpBB/phpbb/error_collector.php [hunk 1]
```diff
@@ -16,15 +16,28 @@
...
-	function __construct()
+	/**
+	 * Constructor.
+	 *
+	 * The variable $error_types may be set to a mask of PHP error types that
+	 * the collector should keep, e.g. `E_ALL`. If unset, the current value of
+	 * the error_reporting() function will be used to determine which errors
+	 * the collector will keep.
+	 *
+	 * @see PHPBB3-13306
```
Comment (Nicofuma) [inline phpBB/phpbb/error_collector.php]: We never include this kind of references, so I'm not sure that we should start

Comment (nickvergessen) [inline phpBB/phpbb/error_collector.php]: From time to time we do (but more in tests), and I don't think it's bad.

Comment (omniError) [inline phpBB/phpbb/error_collector.php]: The tracker page has a longer explanation of the reason for this, I see other projects do it that way and didn't know if there was a style convention for this.

**Pull request merged**